Development

goktugberkulu opened the pull request.

goktugberkulu committed:
Initialize project

goktugberkulu committed:
Create url extension for prism and url builder

goktugberkulu committed:
Add building prism url option with setters instead of url extension

goktugberkulu committed:
Refactor podspec file

taylanpince commented:
Looks pretty good to me. @sakkaras Can you also take a look and approve if it's good to go?

sakkaras commented on Prism-iOS.podspec:
```diff
@@ -0,0 +1,19 @@
+  s.platform = :ios
+  s.ios.deployment_target = '8.0'
+  s.name         = "Prism-iOS"
+  s.version      = "0.0.1"
```
I think we can set version as 1.0.0

sakkaras commented on Prism-iOS.podspec:
```diff
@@ -0,0 +1,19 @@
+  s.ios.deployment_target = '8.0'
+  s.name         = "Prism-iOS"
+  s.version      = "0.0.1"
+  s.summary      = "Prism image resizer iOS library."
```
I think the library is more about building prism urls, not image resizing.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    case topLeft
+    case center
+    case none
+}
```
1- case none should be the first case in enum definitions. 
2- none is not the word to define default case for some enums. We can use undetermined for PrismOutputImageQuality and PrismOutputImageType instead.
3- If we use this library in Objective-C, we should extend enum from Int, otherwise Objective-C does not recognize these enum values. If we do not, then we can directly set values for enum cases instead of using separate methods such as setPrismOutputImageQuality etc.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+
+public class PrismURL {
+    
+    var baseURL: URL? = nil
```
No need to set nil if the variable is optional.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+public class PrismURL {
+    
+    var baseURL: URL? = nil
+    var quality: PrismOutputImageQuality = PrismOutputImageQuality.none
```
No need to explicitly set the variable type when we set a default value. 

var quality = PrismOutputImageQuality.none is enough.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var baseURL: URL? = nil
+    var quality: PrismOutputImageQuality = PrismOutputImageQuality.none
+    var expectedHeight: CGFloat = 0.0
+    var expectedWidth: CGFloat = 0.0
```
var expectedHeight = CGFloat(0.0) // i think this assignment is more appropriate for swift.

sakkaras replied:
The more appropriate solution may be to use CGSize instead of setting a width of height explicitly. If user wants to set one of them, then he/she should use something like 
       UIKIT_EXTERN const CGFloat UIViewNoIntrinsicMetric NS_AVAILABLE_IOS(6_0); // -1.
Also, I think 0 should be an invalid value. If the user sends it, we either return nil as prism URL or we should not add it as a query parameter(Unless prism API requires a value).

One more thing, if the user sets resize mode as none and a specific size, what should be the correct behaviour?

1- If resize mode has higher priority, we should not set size and return the original image.
2- If the size has higher priority, we should change none with a more appropriate word since none has no meaning for the concept.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var expectedWidth: CGFloat = 0.0
+    var resizeMode: PrismOutputImageResizeMode = PrismOutputImageResizeMode.none
+    var cropRect: CGRect = CGRect.zero
+    var imageType: PrismOutputImageType = PrismOutputImageType.none
```
No need to explicitly set the variable type when we set a default value.

sakkaras replied:
I think cropRect variable should be optional. CGRect.zero is an invalid value. This is misleading. This should also be documented.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var cropRect: CGRect = CGRect.zero
+    var imageType: PrismOutputImageType = PrismOutputImageType.none
+    var isPreservingRatio: Bool? = nil
+    var isPremultiplied: Bool? = nil
```
No need to set nil if the variable is optional.

sakkaras replied:
Unless it is really required, I do not think we use optional variables for value types like boolean.  Are you sure?

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var imageType: PrismOutputImageType = PrismOutputImageType.none
+    var isPreservingRatio: Bool? = nil
+    var isPremultiplied: Bool? = nil
+    var gravity: PrismOutputGravity = .none
```
No need to explicitly set the variable type when we set a default value.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var isPreservingRatio: Bool? = nil
+    var isPremultiplied: Bool? = nil
+    var gravity: PrismOutputGravity = .none
+    var frameBackgroundColor: String? = nil
```
No need to set nil if the variable is optional.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+    var isPremultiplied: Bool? = nil
+    var gravity: PrismOutputGravity = .none
+    var frameBackgroundColor: String? = nil
+    var isRetina: Bool? = true
```
Unless it is really required, I do not think we use optional variables for value types like boolean. And here you used true as the default value. Are you sure?

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        }
+        
+        if let query = url.query, query != "" {
+            return url
```
query.isEmpty is more appropriate than ' != "" ' check.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        default:
+            let qualityQueryItem = URLQueryItem(name: "quality", value: setPrismOutputImageQuality())
+            queryParameters.append(qualityQueryItem)
+        }
```
If only case none will be ignored, switch seems weird :). We can use if check here, I suppose.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        default:
+            let gravityQueryItem = URLQueryItem(name: "gravity", value: setPrismOutputGravity())
+            queryParameters.append(gravityQueryItem)
+        }
```
Same as above.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        
+        let stringValueOfWidth = "\(Int(expectedWidth))"
+        return stringValueOfWidth
+    }
```
We should change method name since it does not set anything.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        
+        let stringValueOfHeight = "\(Int(expectedHeight))"
+        return stringValueOfHeight
+    }
```
We should change method name since it does not set anything.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+        default:
+            return nil
+        }
+    }
```
We should change method name since it does not set anything. As I wrote above, it is more appropriate to set these values directly to enum cases if possible.

sakkaras commented on Prism/Prism/Classes/PrismURL.swift:
```diff
@@ -0,0 +1,312 @@
+            return false
+        }
+        return true
+    }
```
This method does not do what its name refers. Hex format should also be checked.

goktugberkulu committed:
Add objective c support for enums and prism url creation, refactor naming and switch cases

goktugberkulu committed:
Update podspec

goktugberkulu committed:
Remove optional init

goktugberkulu committed:
Add documentation and refactor grouping

Added documentation in code.
Separated enums to new files.
Separated extensions to new files.
Refactored grouping.
Added new default values to prism configurations.

goktugberkulu committed:
Merge branch 'master' into development

goktugberkulu committed:
Update documentation
